R/runScenarios.R

```diff
@@ -0,0 +1,171 @@
+          warning(cond)
+        }
+      )
+    }
+  }
+
+  # Simulate all simulations concurrently
+  simulationResults <- runSimulations(simulations = simulations, simulationRunOptions = scenarioConfiguration$simulationRunOptions)
+
+  returnList <- vector("list", length(simulationResults))
+  names(returnList) <- scenarioNames
+  for (idx in seq_along(scenarioNames)) {
+    simulationName <- scenarioNames[[idx]]
+    simulation <- simulations[[simulationName]]
+    if (ospsuite.utils::objectCount(simulationResults) > 1){
+      results <- simulationResults[[simulation$id]]
+    } else {
+      results <- simulationResults
+    }
+    outputValues <- getOutputValues(results,
+                                    quantitiesOrPaths = getAllQuantitiesMatching(enumValues(OutputPaths), simulation)
+    )
+    returnList[[simulationName]] <- list(
+      simulation = simulation, results = results,
+      outputValues = outputValues
+    )
+  }
+
+  return(returnList)
+}
+
+#' Initialize a simulation based on scenario definition
+#'
+#' @description
+#' Load simulation
+#' Apply parameters from global XLS
+#' Apply individual physiology
+#' Apply individual model parameters
+#' Apply test parameters (TestParameters.R)
+#' Set simulation outputs (OutputPaths.R)
+#' Set simulation time
+#' esqlabsR::initializeSimulation()
+#'
+#' @param scenarioConfiguration A \code{ScenarioConfiguration} object
+#' @param customParams A list with three vectors named `paths`, `values`, `units`
+#' to be applied to the model
+#'
+#' @return Initialized \code{Simulation} object
+#' @export
+initializeScenario <- function(scenarioConfiguration, customParams = NULL) {
+  # Update `ScenarioConfiguration` with information from excel
+  scenarioConfiguration <- readScenarioConfigurationFromExcel(scenarioConfiguration)
+  # Read parameters from the parameters file
+  params <- esqlabsR::readParametersFromXLS(
+    file.path(
+      scenarioConfiguration$projectConfiguration$paramsFolder,
+      scenarioConfiguration$projectConfiguration$paramsFile
+    ),
+    scenarioConfiguration$paramSheets
+  )
+
+  individualCharacteristics <- NULL
+  if (!is.null(scenarioConfiguration$individualId)) {
+    individualCharacteristics <- readIndividualCharacteristicsFromXLS(XLSpath = file.path(scenarioConfiguration$projectConfiguration$paramsFolder, scenarioConfiguration$projectConfiguration$individualPhysiologyFile), individualId = scenarioConfiguration$individualId, nullIfNotFound = TRUE)
+
+    if (is.null(individualCharacteristics)){
+      warning(paste0("No individual characteristics for individual id '",
+                     scenarioConfiguration$individualId, "' found."))
+    }
+
+    # Find individual-specific model parameters
+    if (scenarioConfiguration$individualId %in% readxl::excel_sheets(path = file.path(
+      scenarioConfiguration$projectConfiguration$paramsFolder,
+      scenarioConfiguration$projectConfiguration$individualParamsFile
+    ))) {
```

Comment:
I think this will be more readable if the following is assigned to a variable

```r
readxl::excel_sheets(path = file.path(
      scenarioConfiguration$projectConfiguration$paramsFolder,
      scenarioConfiguration$projectConfiguration$individualParamsFile
    ))
```

and then that variable is used in the `if()` condition.